test/optimizers.py

```diff
@@ -0,0 +1,70 @@
+import test_utils
+from torch.optim import SGD
+
+MIN_DELTA_TO_UPDATE_LR = 0.000001
+NUM_WARMUP_EPOCHS = 1
+
+
+class WarmupAndExponentialDecaySGD(SGD):
```

Comment:
I think this might be a good approach, though this should be a wrapper optimizer.
It should take an optimizer as constructor argument, and not subclass SGD.
Otherwise if you want to add LR schedule to another optimizer class, you will have to duplicate the work.